File: Elitesportweb/elite_sports_flutter/lib/models/diet_plan.dart
class DietPlan {
  final String id;
  final String planName;
  final String description;
  final String assignedBy;
  final String? calories;
  final String? duration;
  final String? notes;
  final List<Meal> meals;

  DietPlan({
    required this.id,
    required this.planName,
    required this.description,
    required this.assignedBy,
    this.calories,
    this.duration,
    this.notes,
    required this.meals,
  });

  factory DietPlan.fromJson(Map<String, dynamic> json) {
    return DietPlan(
      id: json['_id'] ?? '',
      planName: json['planName'] ?? '',
      description: json['description'] ?? '',
      assignedBy: json['assignedBy'] ?? 'Instructor',
      calories: json['calories']?.toString(),
      duration: json['duration']?.toString(),
      notes: json['notes'],
      meals: (json['meals'] as List? ?? [])
          .map((meal) => Meal.fromJson(meal))
          .toList(),
    );
  }
}

class Meal {
  final String name;
  final String? time;
  final List<String> foods;

  Meal({
    required this.name,
    this.time,
    required this.foods,
  });

  factory Meal.fromJson(Map<String, dynamic> json) {
    return Meal(
      name: json['name'] ?? '',
      time: json['time'],
      foods: List<String>.from(json['foods'] ?? []),
    );
  }
}

File: Elitesportweb/models/DietPlan.js
import mongoose from 'mongoose'

const DietPlanSchema = new mongoose.Schema({
  memberId: { type: mongoose.Schema.Types.ObjectId, ref: 'Member', required: true },
  instructorId: { type: mongoose.Schema.Types.ObjectId, ref: 'Instructor', required: true },
  title: { type: String, required: true },
  description: { type: String },
  meals: [{
    name: { type: String, required: true },
    time: { type: String, required: true },
    foods: [{ type: String, required: true }],
    calories: { type: Number },
    notes: { type: String }
  }],
  totalCalories: { type: Number },
  duration: { type: String }, // e.g., "4 weeks"
  goals: [{ type: String }],
  restrictions: [{ type: String }],
  status: { type: String, default: 'active', enum: ['active', 'completed', 'paused'] },
  approvalStatus: { type: String, default: 'pending', enum: ['pending', 'approved', 'rejected'] },
  approvedBy: { type: mongoose.Schema.Types.ObjectId, ref: 'Admin' },
  approvedAt: { type: Date },
  rejectionReason: { type: String }
}, { timestamps: true })

export default mongoose.models.DietPlan || mongoose.model('DietPlan', DietPlanSchema)